File: apps/admin-v2/src/application/use-cases/services/get-services.use-case.ts
import type {
  ServiceRepository,
  ListServicesParams,
} from '@/domain/repositories/service.repository';

export class GetServicesUseCase {
  constructor(private repo: ServiceRepository) {}

  execute(params?: ListServicesParams) {
    return this.repo.getAll(params);
  }
}

File: apps/admin_mobile/lib/application/use_cases/services/get_services_use_case.dart
import '../../../domain/entities/service.dart';
import '../../../domain/repositories/service_repository.dart';
import '../../../shared/types/paginated_result.dart';

class GetServicesUseCase {
  final ServiceRepository _repo;
  GetServicesUseCase(this._repo);

  Future<PaginatedResult<Service>> call({int? page}) =>
      _repo.getAll(page: page);
}
